src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/RenameGroupForm.tsx

```diff
@@ -0,0 +1,172 @@
+import { cn } from '@/lib/tailwind/utils';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import React, { FC, KeyboardEvent, useEffect, useRef } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { ContainerCount } from './ContainerCount';
+import { UserCount } from './UserCount';
+
+interface IRenameGroupFormProps {
+  currentGroupName: string;
+  containerCount: number;
+  userCount: number;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const RenameGroupForm: FC<IRenameGroupFormProps> = ({
+  currentGroupName,
+  isOpen,
+  onClose,
+  containerCount,
+  userCount,
+}) => {
+  // input ref
+  const inputRef = useRef<HTMLInputElement>(null);
+  // ref to check if the onBlur event of the input should be handled or not
+  const shouldHandleInputBlur = useRef(true);
+
+  const formSchema = z.object({
+    groupName: z.string().min(1, { message: 'Name is required' }),
+  });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      groupName: currentGroupName,
+    },
+  });
+
+  /**
+   * The type of the form inputs based on the schema
+   */
+  type Inputs = z.infer<typeof formSchema>;
+
+  /**
+   * Process the form submission.
+   * @param values - The form values
+   */
+  const processSubmit: SubmitHandler<Inputs> = async (values) => {
+    const { groupName } = values;
+    /**Replace console.log with rename api method in here, after it's implemented */
+    alert(`Rename to ${groupName}`);
+    form.reset();
+    onClose();
+  };
+
+  /**
+   * Handle the escape key down event.
+   * @param event - The keyboard event
+   */
+  const handleEscKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      onClose();
+    }
+  };
+
+  /**
+   * When the input is blurred,
+   * process the form submission and close the form.
+   */
+  const handleInputBlur = () => {
+    // if the input is blurred because of the cross button click, don't process the form submission.
+    if (!shouldHandleInputBlur.current) return;
+    // process the form submission
+    processSubmit(form.getValues());
+    onClose();
+  };
+
+  /**
+   * When cross button is mouse downed,
+   * set the shouldHandleBlur to false,
+   * and empty the input value.
+   */
+  const handleCrossButtonMouseDown = () => {
+    shouldHandleInputBlur.current = false;
+    form.setValue('groupName', '');
+  };
+
+  /**
+   * When cross button click is completed,
+   * set the focus back to the input,
+   * and set the shouldHandleBlur to true.
+   */
+  const handleCrossButtonClick = () => {
+    inputRef.current?.focus();
+    shouldHandleInputBlur.current = true;
+  };
+
+  /**
+   * Reset the form when the form is closed.
+   */
+  useEffect(() => {
+    if (!isOpen) form.reset();
+  }, [isOpen, form]);
+
+  /**
+   * Focus the input when the form is opened.
+   */
+  useEffect(() => {
```

Comment:
Whenever you face a problem caused by Radix, the first thing to do to resolve it is to explore its official document.
In there, you will find the `onCloseAutoFocus` and notice the right solution.

See: https://www.radix-ui.com/primitives/docs/components/dropdown-menu#content

Reply:
In our verbal discussion, we decided to not use `onCloseAutoFocus` as it affects other cases, such as closing the dropdown menu by pressing the [ESK] key, which is a side effect.

Reply:
Thank you for researching!
After applied UI from wireframe correctly, this problem went away because the menu button is not displayed when form is open.